### PR TITLE
Better API for Triggers - WithTrigger(event, detail, timing)

### PR DESCRIPTION
**Pull request by rudiv**
I didn't expect to do 2 PRs in one day but here it is 🥳 My development team has been using this library with their more recent builds and the biggest grumble they have is what follows, so good work and thank you for building something that is just right for working with HTMX!

The current API for triggers (`.Trigger(value)`) provides the ability to set the header once, meaning that in its most basic form a trigger with no parameters can be specified:
```c#
h.Trigger("event-name");
```

For more control, a manual string would have to be constructed:
```c#
h.Trigger("{\"event-name\":{myVal:42,anotherVal:12}}");
```

For multiple events, the interaction only gets more complex and the user would typically resort to manually constructing JSON or serialising a dictionary:
```c#
h.Trigger("{\"event-name\":{myVal:42,anotherVal:12},\"another-event\":{what:false}}");
```

With that I propose a new API for adding triggers to the response which does the dictionary serialisation method internally, the above details with parameters could now be rewritten as follows:
```c#
h.WithTrigger("event-name", new { myVal = 42, anotherVal = 12 })
   .WithTrigger("another-event", new { what = false });
```

Of course this works the same with or without detail being provided for the event, and timing can also be overridden (to push to an alternative header):
```c#
h.WithTrigger("event-name", timing: HtmxTriggerTiming.AfterSettle);
```

Notes:

- The Readme has been updated with this (and also a note about the Antiforgery tag, but feel free to remove that)
- The header builder, if there is a single event with no params, will only output the event name (rather than full JSON) in the spirit to save a few characters. Every little helps!
- There are tests that cover off all scenarios, I think, so let me know if you need to see more

**Major note:**

I built this against .NET 6, rather than 5.0. I appreciate it's a big change (and the Action will need updating) however given that 5 is out of support it's probably worth switching to the LTS. (That's why the action is currently failing)

### CODE REVIEW

**Comment by khalidabuhakmeh** [comment]
I like this idea. Let me look at updating the action, so it builds successfully.

**Review by khalidabuhakmeh — APPROVED** [review]
LGTM! Getting @lxztlr to do another pass, but I like the idea.

**Comment by rudiv** [comment]
Glad you approve - we're actively using it and it just feels natural 👍

Fairly primitive change but happy to make any amends.

File: src/Htmx/HtmxResponseHeaders.cs [hunk 1]
```diff
@@ -153,5 +161,78 @@ public HtmxResponseHeaders Retarget(string value)
...
+                if (_headers.ContainsKey(Keys.Trigger))
+                {
+                    throw new Exception("You must use either WithTrigger(..) or Trigger(..), but not both.");
+                }
+                
+                _headers[Keys.Trigger] = BuildTriggerHeader(HtmxTriggerTiming.Default);
+            }
+            if (_triggers.ContainsKey(HtmxTriggerTiming.AfterSettle))
+            {
+                if (_headers.ContainsKey(Keys.TriggerAfterSettle))
```
Comment (AlexZeitler) [inline src/Htmx/HtmxResponseHeaders.cs:203]: Maybe add some tests for the unhappy paths as well?

**Review by AlexZeitler — COMMENTED** [review]
LGTM aside the tests for unhappy paths.

File: src/Htmx/HtmxResponseHeaders.cs (hunk 1 above)
Comment (rudiv) [reply src/Htmx/HtmxResponseHeaders.cs:203]: Consider it done

**Comment by rudiv** [comment]
Tests added to detect usage of legacy / new APIs together (which wouldn't work as they would override each other)

**Review by AlexZeitler — APPROVED** [review]

**Comment by khalidabuhakmeh** [comment]
I will merge and deploy this tomorrow. Should I bump this package up to v1.0 at this point?

**Comment by AlexZeitler** [comment]
Feels stable to me, so v1 makes sense.

**Comment by rudiv** [comment]
There's not much missing from the surface either, so it's probably a good candidate now that it's aligned with the latest version of HTMX.

We added another couple of helpers internally (things such as a new tag helper outputting elements that request via HTMX on load) that may be useful, but I may propose later if they prove useful and general purpose.

**Comment by khalidabuhakmeh** [comment]
LFG!!!!!!!

**Comment by khalidabuhakmeh** [comment]
In the subsequent releases, I'll start removing the obsolete methods. Thanks, everyone. The package in its v1.0.0 should be out on NuGet soon.

**Pull request merged**